File: app/src/main/java/me/wcy/music/enums/PlayModeEnum.java
package me.wcy.music.enums;

/**
 * 播放模式
 * Created by wcy on 2015/12/26.
 */
public enum PlayModeEnum {
    LOOP(0),
    SHUFFLE(1),
    SINGLE(2);

    private int value;

    PlayModeEnum(int value) {
        this.value = value;
    }

    public static PlayModeEnum valueOf(int value) {
        switch (value) {
            case 1:
                return SHUFFLE;
            case 2:
                return SINGLE;
            case 0:
            default:
                return LOOP;
        }
    }

    public int value() {
        return value;
    }
}

File: app/src/main/java/me/wcy/music/service/PlayModeEnum.kt
package me.wcy.music.service

/**
 * 播放模式
 * Created by wcy on 2015/12/26.
 */
enum class PlayModeEnum(private val value: Int) {
    LOOP(0),
    SHUFFLE(1),
    SINGLE(2);

    fun value(): Int {
        return value
    }

    companion object {
        fun valueOf(value: Int): PlayModeEnum {
            return when (value) {
                1 -> SHUFFLE
                2 -> SINGLE
                0 -> LOOP
                else -> LOOP
            }
        }
    }
}